### PR TITLE
refactor(DA): refine DA stream define and implementation

### PR DESCRIPTION
**Pull request by popcnt1**
## Summary

1. clarify stream, batch, l2 block, chunk ,segment's relations
2. implement necessary methods of chunk and segment
3. refine code logic of open-da/celestia submit
4. refine mainet deploy workflow

### CODE REVIEW

File: .github/workflows/deploy_mainnet.yml [hunk 1]
```diff
@@ -26,5 +26,11 @@ jobs:
           chmod 600 private_key.pem
           sudo apt update
           sudo apt install -y --no-install-recommends openssh-server
-          ssh -o StrictHostKeyChecking=no -i private_key.pem $USER@$HOST bash -c "'sleep 30' && && docker image prune -a -f && docker ps | grep rooch | awk '{print \$1}' | xargs -r docker stop && docker ps -a | grep rooch | awk '{print \$1}' | xargs -r docker rm -f && docker pull 'ghcr.io/rooch-network/rooch:${{ github.event.inputs.tagName }}' && docker run -d -v /data:/root -p 6767:6767 'ghcr.io/rooch-network/rooch:${{ github.event.inputs.tagName }}' server start -n main --btc-rpc-url '${{secrets.BTC_MAIN_RPC_URL}}' --btc-rpc-username rooch-main --btc-rpc-password '${{secrets.BTC_MAIN_RPC_PWD}}' --btc-start-block-height 0 --btc-end-block-height 767420 --data-import-mode 1"
-
+          BRANCH=$(basename ${{ github.ref }})
+          ssh -o StrictHostKeyChecking=no -i private_key.pem $USER@$HOST \
+            "cd /root/rooch && git fetch origin && git checkout -B $BRANCH origin/$BRANCH || git checkout $BRANCH && bash scripts/deploy_rooch_testnet.sh \
```
Comment (jolestar) [inline .github/workflows/deploy_mainnet.yml]: `deploy_rooch_testnet.sh` -> `deploy_rooch_mainnet.sh`

**Review by jolestar — APPROVED** [review]

**Review by yourmoonlight — APPROVED** [review]
LGTM

File: scripts/deploy_rooch_mainnet.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+REF="$1"
+BTC_MAIN_RPC_URL="$2"
+BTC_MAIN_RPC_PWD="$3"
+OPENDA_GCP_MAINNET_BUCKET="$4"
+OPENDA_GCP_MAINNET_CREDENTIAL="$5"
+
+sleep 30
+docker image prune -a -f
+docker ps | grep rooch | awk '{print $1}' | xargs -r docker stop
```
Comment (baichuan3) [inline scripts/deploy_rooch_mainnet.sh:13]: 这样会把 rooch 其它进程杀掉，比如 rooch-faucet ？

Comment (popcnt1) [reply scripts/deploy_rooch_mainnet.sh:13]: 主网节点应该不会跑 faucet？或者我把 grep -v faucet 加进去也行

**Pull request merged**